packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -3597,6 +3617,59 @@ export class SnapController extends BaseController<
         }
         return result;
       }
+      case HandlerType.OnAssetsLookup: {
+        // We know the permissions are guaranteed to be set here.
```

Comment:
Maybe extract into separate functions.

Reply:
I was thinking of doing something like that, but it feels like it might be a larger refactor that is out of scope now. 

Generally I think we should look at how we can move logic out of the SnapController and/or genericize some of the existing logic. The `assertSnapRpcRequestResult` function is another good one.

Reply:
I was thinking for now we could just create two functions, one for each handler, which contains the logic? Unless I misunderstand something, that shouldn't require any refactors?

Reply:
Do you mean inside the class or outside? 🤔

Reply:
Either is fine with me. Whatever is easiest.